### DISCUSSION TITLE
FundMeTest : Question about Generated Address in Test Cases

### DISCUSSION THREAD

**Discussion by devrajsinghrawat** [Cyfrin/foundry-full-course-cu#211] [Q&A] [answered]
## 

Hello Everyone,

If you are following the 2nd part of the course where we are writing the test cases for FundMe Contract, you may have noticed that the `msg.sender` (i.e., the externally owned account) is set to "0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38" for these test cases.

Ref : https://youtu.be/sas02qSFZ74?t=1375

I have reviewed the list of 10 generated addresses in the foundry and noticed that this address is not part of these addresses:

(0) 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
(1) 0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d
(2) 0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a
(3) 0x7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6
(4) 0x47e179ec197488593b187f80a00eb0da91f1b9d0b13f8733639f19c30a34926a
(5) 0x8b3a350cf5c34c9194ca85829a2df0ec3153be0318b5e2d3348e872092edffba
(6) 0x92db14e403b83dfe3df233f83dfa3a0d7096f21ca9b0d6d6b8d88b2b4ec1564e
(7) 0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356
(8) 0xdbda1821b80551c9d65939329250298aa3472ba22feea921c0cf5d620ea67b97
(9) 0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6

**Has anyone been able to determine the source or origin of the address "0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38"? It seems to be unrelated to the generated addresses provided in the foundry.**

Thank you for your assistance!

**Comment by ghost** [comment, marked as answer]
Hello @devrajsinghrawat you can find the answer in [this section](https://book.getfoundry.sh/reference/config/testing#sender) of the docs. That address is both the default `msg.sender` and `tx.origin` in the tests.

**Reply by devrajsinghrawat** [reply]
Ahh,, Thanks it seems default foundry config setting for test enviorment..
